test/e_chronicler/models/journal_entry_test.exs

```diff
@@ -66,5 +68,26 @@ defmodule EChronicler.Models.JournalEntryTest  do
     refute invalid_changeset.valid?
   end
 
+  test "change_journal_entry/1 returns a journal_entry changeset" do
+    assert {:ok, %JournalEntry{} = journal_entry} = EChronicler.Models.JournalEntry.create_journal_entry(@valid_journal_entry)
+
+    assert %Ecto.Changeset{} = JournalEntry.change_journal_entry(journal_entry)
+  end
+
+  test "update_journal_entry/2 with valid data updates the journal_entry" do
+    assert {:ok, %JournalEntry{} = journal_entry} = EChronicler.Models.JournalEntry.create_journal_entry(@valid_journal_entry)
+    update_attrs = %{author: "some updated author", entry: "some updated entry", title: "some updated title"}
+
+    assert {:ok, %JournalEntry{} = journal_entry} = JournalEntry.update_journal_entry(journal_entry, update_attrs)
+    assert journal_entry.author == "some updated author"
+    assert journal_entry.entry == "some updated entry"
+    assert journal_entry.title == "some updated title"
+  end
+
+  test "update_journal_entry/2 with invalid data returns error changeset" do
+    assert {:ok, %JournalEntry{} = journal_entry} = EChronicler.Models.JournalEntry.create_journal_entry(@valid_journal_entry)
+    assert {:error, %Ecto.Changeset{}} = JournalEntry.update_journal_entry(journal_entry,  @invalid_journal_entry)
+    assert journal_entry == JournalEntry.get_journal_entry(journal_entry.id)
```

Comment:
Not sure what the purpose of this last line is...?

But overall, I really think your elixir skills are getting super charged.  Nice work.

Reply:
Deleted for clarity! It seemed off topic.